observation_portal/common/rise_set_utils.py

```diff
@@ -340,7 +342,24 @@ def get_site_rise_set_intervals(start, end, site_code):
     return []
 
 
-def is_interval_available_for_telescope(start: datetime, end: datetime, site: str, enclosure: str, telescope: str) -> bool:
+def realtime_intervals_to_block_for_telescope(start: datetime, end: datetime, site: str, enclosure: str, telescope: str) -> Intervals:
+    """Returns an Intervals object containing time intervals to block out realtime observing, per telescope.
+       This is meant to be overridden in a custom Observation Portal to add whatever rules are desired for blocking
+       realtime sessions.
+       
+    Parameters:
+        start: The start time
+        end: The end time
+        site: The site code
+        enclosure: The enclosure code
+        telescope: The telescope code
+    Returns:
+        Intervals object of time intervals to block realtime sessions.
+    """
+    return Intervals()
+
+
+def is_realtime_interval_available_for_telescope(start: datetime, end: datetime, site: str, enclosure: str, telescope: str) -> bool:
```

Comment:
This method is called when validating a user booking a specific realtime session to make sure its a valid time.